nix/shells/rust-backend.nix

```diff
@@ -0,0 +1,63 @@
+{
+  pkgs,
+  inputs',
+  self',
+}: let
+  ourPkgs = self'.packages;
+in
+  with pkgs;
+    mkShell {
+      packages = [
+        # Rust toolchain
+        cargo
+        rustfmt
+        clippy
+        rust-analyzer
+
+        # Database tools
+        sqlite
+
+        # Development tools
+        just
+        git
+
+        # Basic utilities
+        coreutils
+        gnugrep
+        gawk
+        ripgrep
+
+        # Build dependencies that Rust projects might need
+        gcc
+        binutils
+        openssl
+        pcre
+        glib
+        libelf
+
+        # For inspecting and debugging
+        pstree
+        hexdump
+        tree-sitter
+
+        # Useful for development
+        vim
+        delta
+      ];
+
+      shellHook = ''
+        echo "🦀 Rust backend development environment"
+        echo "Working directory: db-backend/"
```

Comment:
we have more rust code: also small-lang, tui, eventually runtime_tracing(but it's a submodule): i wonder if we should name this more generally `rust-shell` or something like this